drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/utils/GitHubLastUpdateCache.java

```diff
@@ -0,0 +1,116 @@
+    public GitHubLastUpdateCache() {
+        this(DEFAULT_CACHE_DIRECTORY);
+    }
+
+    private File createCacheDirectory(File cacheDirectory) {
+        if (cacheDirectory.exists() && !cacheDirectory.isDirectory()) {
+            throw new IllegalArgumentException("Passed [" + cacheDirectory.getAbsolutePath() + "] exists and is not a directory.");
+        }
+        if (!cacheDirectory.exists()) {
+            cacheDirectory.mkdir();
+        }
+        return cacheDirectory;
+    }
+
+    public ZonedDateTime lastModificationOf(String uniqueKey) {
+        final ZonedDateTime lastModification;
+        if (Files.exists(Paths.get(createCachedFilePath(uniqueKey)))) {
+            final JsonElement lastModificationDate = deserializeCachedFile(uniqueKey).get(LAST_MODIFIED_PROPERTY);
+            lastModification = gson.fromJson(lastModificationDate, ZonedDateTime.class);
+        } else {
+            lastModification = ZonedDateTime.of(2008, 4, 10, 0, 0, 0, 0, ZoneId.of("GMT"));
+        }
+        return lastModification;
+    }
+
+    public <T> T load(String uniqueKey, Class<T> type) {
+        final JsonObject asset = deserializeCachedFile(uniqueKey).getAsJsonObject(ASSET_PROPERTY);
+        return gson.fromJson(asset, type);
+    }
+
+    public <T> void store(T asset, String uniqueKey, ZonedDateTime dateTime) {
+        final String cachedFilePath = createCachedFilePath(uniqueKey);
+        final JsonObject jsonObject = combineAsJson(asset, dateTime);
+        try (final FileOutputStream fileOutputStream = new FileOutputStream(cachedFilePath, false)) {
+            fileOutputStream.write(jsonObject.toString().getBytes());
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to store " + jsonObject + "%n as [" + cachedFilePath + "]", e);
+        }
+    }
+
+    private String createCachedFilePath(String uniqueKey) {
+        return cacheDirectory.getAbsolutePath() + "/gh.cache." + uniqueKey + ".json";
+    }
+
+    private JsonObject deserializeCachedFile(String uniqueKey) {
+        final String cachedFilePath = createCachedFilePath(uniqueKey);
+        try (final FileReader reader = new FileReader(cachedFilePath)) {
+            return gson.fromJson(reader, JsonObject.class);
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to deserialize file [" + cachedFilePath + "]", e);
+        }
+    }
+
+    private <T> JsonObject combineAsJson(T asset, ZonedDateTime dateTime) {
+        final JsonElement assetAsJson = gson.toJsonTree(asset);
+        final JsonObject jsonObject = new JsonObject();
+        jsonObject.add(LAST_MODIFIED_PROPERTY, gson.toJsonTree(dateTime));
+        jsonObject.add(ASSET_PROPERTY, assetAsJson);
+        return jsonObject;
+    }
+
+    private static class ZonedDateTimeConverter implements JsonSerializer<ZonedDateTime>, JsonDeserializer<ZonedDateTime> {
+        private static final DateTimeFormatter FORMATTER = DateTimeFormatter.RFC_1123_DATE_TIME;
```

Comment:
The date should be in format RFC2616:
`"EEE, dd MMM yyyy HH:mm:ss zzz"`
otherwise the rate limit is still counted even when 304 is returned. (probably bug in github api)
The difference is:
RFC1123: `Thu, 9 Mar 2017 21:20:23 GMT`
RFC2626: `Thu, 09 Mar 2017 21:20:23 GMT`